beacon-chain/node/node.go

```diff
@@ -674,8 +676,10 @@ func (b *BeaconNode) registerRPCService() error {
 		EnableDebugRPCEndpoints: enableDebugRPCEndpoints,
 		MaxMsgSize:              maxMsgSize,
 
-		// Vanguard: un-confirmed cached block fetcher
+		// vanguard: EnableVanguardNode and UnconfirmedBlockFetcher is used for vanguard chain
+		EnableVanguardNode:      b.cliCtx.Bool(cmd.VanguardNetwork.Name),
```

Comment:
I'd suggest to rename it. IMHO vanguard is more like a mode, so maybe `VanguardModeEnabled`? Also I'd suggest to use past tense in such flags